cmd/ig/daemon.go

```diff
@@ -87,11 +116,41 @@ func newDaemonCommand(runtime runtime.Runtime) *cobra.Command {
 
 		log.Infof("starting Inspektor Gadget daemon at %q", socket)
 		service := gadgetservice.NewService(log.StandardLogger(), eventBufferLength)
+
+		var options []grpc.ServerOption
+
+		if serverKey != "" || serverCert != "" || clientCA != "" {
+			cert, err := tls.LoadX509KeyPair(serverCert, serverKey)
+			if err != nil {
+				return fmt.Errorf("loading TLS keypair: %w", err)
+			}
+
+			ca := x509.NewCertPool()
+			caBytes, err := os.ReadFile(clientCA)
+			if err != nil {
+				return fmt.Errorf("loading client CA certificate: %w", err)
+			}
+
+			if ok := ca.AppendCertsFromPEM(caBytes); !ok {
+				return errors.New("failed to parse client CA certificate")
+			}
+
+			tlsConfig := &tls.Config{
+				ClientAuth:   tls.RequireAndVerifyClientCert,
+				Certificates: []tls.Certificate{cert},
+				ClientCAs:    ca,
+			}
+
+			options = append(options, grpc.Creds(credentials.NewTLS(tlsConfig)))
+		} else {
+			log.Warnf("no TLS configuration provided, running in insecure mode")
```

Comment:
Do we need to print this message even when unix sockets are used? I'd argue it's not needed as in that case the client needs access to the unix socket file to contact the server.